_posts/2018-10-25-Rust-1.30.md

```diff
@@ -40,18 +40,19 @@ struct Pet {
 }
 ```
 
-And convert a `Pet` to and from JSON.
+And convert a `Pet` to and from JSON because `serde_json` defined `Serialize` and
+`Deserialize` in a procedural macro.
 
 Rust expands on this by adding the ability to define two other kinds of
 advanced macros, "attribute-like procedrual macros" and "function-like
 procedural macros."
 
-Attribute-like macros are similar to custom derive macros, but instead of
-generating code for `#[derive]`, they allow you to create new, custom
-attributes of your own. They're also more flexible; derive only works for
-structs and enums; attributes can go on other places as well, like functions.
-As an example of using an attribute-like macro, you might have something like
-this when using a web application framework:
+Attribute-like macros are similar to custom derive macros, but instead of generating code
```

Comment:
Wording nit: attribute macros are not "-like" attributes, they *are* attributes.

(While function-like macros are obviously not functions, so the "-like" suffix is justified there.)